x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_primary_navigation.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { getOr } from 'lodash/fp';
+import { useEffect, useState, useCallback } from 'react';
+import { i18n } from '@kbn/i18n';
+
+import { PrimaryNavigationProps } from './types';
+import { usePrimaryNavigationItems } from './use_navigation_items';
+import { KibanaPageTemplateProps } from '../../../../../../../../src/plugins/kibana_react/public';
+
+const translatedNavTitle = i18n.translate('xpack.securitySolution.navigation.mainLabel', {
+  defaultMessage: 'Security',
+});
+
+export const usePrimaryNavigation = ({
+  filters,
+  query,
+  navTabs,
+  pageName,
+  savedQuery,
+  sourcerer,
+  tabName,
+  timeline,
+  timerange,
+}: PrimaryNavigationProps): KibanaPageTemplateProps['solutionNav'] => {
+  const mapLocationToTab = useCallback(
```

Comment:
you do not need to use `useCallback` here, it does not matter if the `mapLocationToTab` reference change.

Reply:
All of this is taken from the tab navigation file to keep in sync with the logic there, but it actually makes sense to have it based on the function being called in the useEffect